packages/grid/_modules_/grid/hooks/utils/useOptionsProp.ts

```diff
@@ -79,6 +80,7 @@ export function useOptionsProp(apiRef: ApiRef, props: GridComponentProps): GridO
       sortModel: props.sortModel,
       density: props.density,
       filterModel: props.filterModel,
+      localeText: { ...DEFAULT_LOCALE_TEXT, ...props.localeText },
```

Comment:
that's done using the `DEFAULT_GRID_OPTIONS` merge

Reply:
in `optionsReducer`

Reply:
Unfortunately it doesn't work. It overwrites the existing prop completely and I don't want that, I want a deep merge.